tests/udp.rs

```diff
@@ -26,6 +27,124 @@ fn udp_peers_not_connected_huge_data_exchange() {
+    const SOCKET1_TOKEN: Token = Token(0);
+    const SOCKET2_TOKEN: Token = Token(1);
+    const SOCKET3_TOKEN: Token = Token(2);
+
+    let mut sock1 = unwrap!(UdpSock::bind(&unwrap!("127.0.0.1:0".parse())));
+    let server_addr = unwrap!(sock1.local_addr());
+    let mut sock2 = unwrap!(UdpSock::bind(&unwrap!("127.0.0.1:0".parse())));
+    let sock2_addr = unwrap!(sock2.local_addr());
+    let mut sock3 = unwrap!(UdpSock::bind(&unwrap!("127.0.0.1:0".parse())));
+    let sock3_addr = unwrap!(sock3.local_addr());
+
+    let poll = unwrap!(Poll::new());
+    unwrap!(poll.register(&sock1, SOCKET1_TOKEN, Ready::readable(), PollOpt::edge()));
+    unwrap!(poll.register(&sock2, SOCKET2_TOKEN, Ready::writable(), PollOpt::edge()));
+    unwrap!(poll.register(&sock3, SOCKET3_TOKEN, Ready::writable(), PollOpt::edge()));
+
+    let mut received_from = Vec::new();
+    let mut events = Events::with_capacity(1024);
+    'event_loop: loop {
+        let _ = unwrap!(poll.poll(&mut events, None));
+        for event in events.iter() {
+            match event.token() {
+                SOCKET1_TOKEN => {
+                    loop {
+                        let res: Option<(Vec<u8>, _)> = unwrap!(sock1.read_frm());
+                        if let Some((_data, peer_addr)) = res {
+                            received_from.push(peer_addr);
+                        } else {
+                            break;
+                        }
+                    }
+                    if received_from.len() == 2 {
+                        break 'event_loop;
+                    }
+                }
+                SOCKET2_TOKEN => {
+                    let data_sent =
+                        unwrap!(sock2.write_to(Some((vec![1u8, 2, 3], server_addr, 1))));
+                    assert!(data_sent);
+                    unwrap!(poll.deregister(&sock2));
+                }
+                SOCKET3_TOKEN => {
+                    let data_sent =
+                        unwrap!(sock3.write_to(Some((vec![1u8, 2, 3], server_addr, 1))));
+                    assert!(data_sent);
+                    unwrap!(poll.deregister(&sock3));
+                }
+                _ => panic!("Unexpected event"),
+            }
+        }
+    }
+
+    assert_that!(
+        &received_from,
+        contains(vec![sock2_addr, sock3_addr]).exactly()
+    );
+}
+
+#[test]
+fn connected_socket_receives_packets_only_from_connected_socket() {
+    const SOCKET1_TOKEN: Token = Token(0);
+    const SOCKET2_TOKEN: Token = Token(1);
+    const SOCKET3_TOKEN: Token = Token(2);
+    const TIMEOUT_TOKEN: Token = Token(3);
+
+    let mut sock2 = unwrap!(UdpSock::bind(&unwrap!("127.0.0.1:0".parse())));
+    let sock2_addr = unwrap!(sock2.local_addr());
+
+    let mut sock1 = unwrap!(UdpSock::bind(&unwrap!("127.0.0.1:0".parse())));
+    unwrap!(sock1.connect(&sock2_addr));
+    let server_addr = unwrap!(sock1.local_addr());
+    unwrap!(sock2.connect(&server_addr));
```

Comment:
is this necessary ? What is it trying to prove ?

Reply:
on line 130 we're using `write()` instead of `write_to()` - in such case the socket must be connected

Reply:
it's not necessary, it just exercises both APIs: write and write_to

Reply:
Na let's keep the test specific in this case. Don't want it to convey wrong ms that both sided connections are somehow necessary. If you want you can do a specific test to test and contrast `write()/write_to()` but let's keep this one showing only the rx side's behaviour that the issue/test-function-name asks for.